api/billingd/service/service.go

```diff
@@ -1019,11 +1030,18 @@ func (s *Service) reportUsage() error {
 }
 
 func (s *Service) reportCustomerUsage(ctx context.Context, cus *Customer) error {
+	deps, err := s.cdb.CountDocuments(ctx, bson.M{"parent_key": cus.Key})
+	if err != nil {
+		return err
+	}
+	summary := s.getSummary(cus, deps)
+
 	for k, usage := range cus.DailyUsage {
 		if product, ok := s.products[k]; ok {
 			if err := s.reportUnits(product, usage, cus.ParentKey); err != nil {
 				return err
 			}
+			addProductToSummary(summary, product, usage.Total)
```

Comment:
split this out as a help function. so instead of getSummary(cus, prod), you just getSummary(cus) and then add any number of products with addProductionToSummary (so here in the loop and earlier one off)

Reply:
Slick